test/framework/seedregistrationframework.go

```diff
@@ -0,0 +1,376 @@
+
+const (
+	kubeconfigString = "kubeconfig"
+	//accesKeyString    = "secretAccessKey"
+	//accessKeyIdString = "accessKeyID"
+)
+
+// SeedRegistrationFramework represents the seed test framework that includes
+// test functions that can be executed on a specific seed
+type SeedRegistrationFramework struct {
+	*GardenerFramework
+	TestDescription
+	Config *SeedRegistrationConfig
+
+	SeedClient kubernetes.Interface
+	Seed       *gardencorev1beta1.Seed
+	Secret     *corev1.Secret //the secret that will be created and used for seed.Spec.SecretRef
+}
+
+// SeedRegistrationConfig is the configuration for a seed registration framework that will be filled with user provided data
+type SeedRegistrationConfig struct {
+	GardenerConfig *GardenerConfig
+	SeedName       string
+	AccessKeyId    string
+	AccessKey      string
+	SecretBinding  string
+	// SeedSecretName        string
+	// SeedSecretNamespace   string
+	// BackupSecretName      string
+	// BackupSecretNamespace string
+	BackupSecretProvider  string
+	ShootedSeedName       string
+	ShootedSeedNamespace  string
+	ShootedSeedKubeconfig string
+	Provider              string
+	ProviderType          string
+	NetworkingDefServices string
+	NetworingDefPods      string
+	NetworingPods         string
+	NetworingServices     string
+	NetworingkNodes       string
+	NetworingkBlockCIDRs  string
+}
+
+// NewSeedRegistrationFramework creates a new simple Seed registration framework
+func NewSeedRegistrationFramework(cfg *SeedRegistrationConfig) *SeedRegistrationFramework {
+	var gardenerConfig *GardenerConfig
+	if cfg != nil {
+		gardenerConfig = cfg.GardenerConfig
+	}
+
+	f := &SeedRegistrationFramework{
+		GardenerFramework: NewGardenerFrameworkFromConfig(gardenerConfig),
+		TestDescription:   NewTestDescription("SEED"),
+		Config:            cfg,
+	}
+
+	CBeforeEach(func(ctx context.Context) {
+		f.CommonFramework.BeforeEach()
+		f.GardenerFramework.BeforeEach()
+		f.BeforeEach(ctx)
+	}, 8*time.Minute)
+	CAfterEach(f.AfterEach, 10*time.Minute)
+	return f
+}
+
+// NewSeedRegistrationFrameworkFromConfig creates a new seed framework from a seed configuration without registering ginkgo
+// specific functions
+func NewSeedRegistrationFrameworkFromConfig(cfg *SeedRegistrationConfig) (*SeedRegistrationFramework, error) {
+	var gardenerConfig *GardenerConfig
+	if cfg != nil {
+		gardenerConfig = cfg.GardenerConfig
+	}
+	f := &SeedRegistrationFramework{
+		GardenerFramework: NewGardenerFrameworkFromConfig(gardenerConfig),
+		TestDescription:   NewTestDescription("SEED"),
+		Config:            cfg,
+	}
+	return f, nil
+}
+
+// BeforeEach should be called in ginkgo's BeforeEach.
+// It sets up the seed framework.
+func (f *SeedRegistrationFramework) BeforeEach(ctx context.Context) {
+	f.Config = mergeSeedConfig(f.Config, seedRegistrationConfig)
+	validateSeedConfig(f.Config)
+}
+
+// AfterEach should be called in ginkgo's AfterEach.
+// Cleans up resources and dumps the shoot state if the test failed
```

Comment:
```suggestion
// Cleans up resources and dumps the seed state if the test failed
```
This should be `seed` state right?